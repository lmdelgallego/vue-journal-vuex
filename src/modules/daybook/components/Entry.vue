<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'daybook-entry', params: { id: 10 } })"
  >
    <!-- Title -->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">15</span>
      <span class="mx-1 fs-5">Julio</span>
      <span class="mx-2 fw-light">2021, jueves</span>
    </div>

    <div class="entry-description">
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Provident velit
      quisquam laboriosam excepturi repudiandae, quae odio rem voluptatum harum
      vero ad! Aspernatur voluptatum quisquam blanditiis dolorum aut doloribus
      quidem similique!
    </div>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in-out;
  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in-out;
  }
  .entry-description {
    font-size: 12px;
  }
}
</style>
