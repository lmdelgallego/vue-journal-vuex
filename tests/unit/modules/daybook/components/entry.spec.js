import {shallowMount} from '@vue/test-utils';
import Entry from '@/modules/daybook/components/Entry.vue';

describe('Entry.vue', () => {

  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(Entry, {
    propsData: {
      entry: {
        id: 1,
        date: 1635539262122,
        text: 'Test content',
      }
    },

    global: {
      mocks: {
        $router: mockRouter,
      },
    },

  });

  test('renders correctly', () => {

    expect(wrapper.element).toMatchSnapshot();
  });

  test('should be redirect correctly to "{name: daybook-entry}" when entry-container is clicked', () => {
    wrapper.find('.entry-container').trigger('click');
    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'daybook-entry', params: { id: 1 } });
  });

  test('should be generate correctly computed properties', () => {
    expect(wrapper.vm.day).toBe(29);
    expect(wrapper.vm.month).toBe('Octubre');
    expect(wrapper.vm.yearDay).toBe('2021, Viernes');
  })

})