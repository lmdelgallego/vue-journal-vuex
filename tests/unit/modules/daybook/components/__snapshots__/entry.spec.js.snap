// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry.vue renders correctly 1`] = `
<div
  class="entry-container mb-3 pointer p-2"
>
  <!-- Title -->
  <div
    class="entry-title d-flex"
  >
    <span
      class="text-success fs-5 fw-bold"
    >
      29
    </span>
    <span
      class="mx-1 fs-5"
    >
      Octubre
    </span>
    <span
      class="mx-2 fw-light"
    >
      2021, Viernes
    </span>
  </div>
  <div
    class="entry-description"
  >
    Test content
  </div>
</div>
`;
